schedview/app/prenight/prenight.py

```diff
@@ -1,418 +1,648 @@
-            # look at the first.
-            end_mjd = observations["mjd"].min() + 1
-
-        end_mjd_almanac = observatory.almanac.get_sunset_info(end_mjd)
-
-        # If the last observation is in the first half (pm) of the night,
-        # guess that we want to look at the night before. If the simulator
-        # is configured to end on an integer mjd, it can happen that that
-        # the start of a night is just after the mjd rollover, so we can
-        # get just a few observations on the last night, and this last
-        # night is probably not the one we want to look at.
-        end_mjd_night_middle = 0.5 * (
-            end_mjd_almanac["sunset"] + end_mjd_almanac["sunrise"]
-        )
-        if end_mjd < end_mjd_night_middle:
-            end_mjd_almanac = observatory.almanac.get_sunset_info(end_mjd - 1)
-
-        # Get the night MJD based on local noon of sunset.
-        sunset_mjd_ut = end_mjd_almanac["sunset"]
-        sunset_mjd_local = sunset_mjd_ut + site.lon.deg / 360
-        sunset_night_mjd = np.floor(sunset_mjd_local)
-        obs_night = Time(sunset_night_mjd, format="mjd", scale="utc")
-
-    night = pn.widgets.DatePicker(name="Night", value=obs_night.datetime.date())
-    timezone = pn.widgets.Select(
-        name="Timezone",
-        options=[
-            "Chile/Continental",
-            "US/Pacific",
-            "US/Arizona",
-            "US/Mountain",
-            "US/Central",
-            "US/Eastern",
-        ],
+    # Use param to extract the sunset time from the almanac table
+    # and convert it no an astropy.time.Time instance
+    # "once and for all", rather than doing it independently everywhere
+    # we need it.
+    _sunset_time = schedview.param.TypedParameter(
+        None,
+        value_type=Time,
+        doc="The time of sunset on the night being examined.",
     )
 
-    scheduler_fname = pn.widgets.TextInput(
-        name="Scheduler file name", value=scheduler_fname
+    # Use param to extract the sunset time from the almanac table
+    # and convert it no an astropy.time.Time instance
+    # "once and for all", rather than doing it independently everywhere
+    # we need it.
+    _sunrise_time = schedview.param.TypedParameter(
+        None,
+        value_type=Time,
+        doc="The time of sunrise on the night being examined.",
     )
-    opsim_output_fname = pn.widgets.TextInput(
-        name="Opsim output", value=opsim_output_fname
+
+    _reward_df = schedview.param.DataFrame(
+        None,
+        columns={
+            "basis_function": str,
+            "feasible": np.bool_,
+            "max_basis_reward": float,
+            "basis_area": float,
+            "basis_weight": float,
+            "tier_label": str,
+            "survey_label": str,
+            "survey_class": str,
+            "survey_reward": float,
+            "basis_function_class": object,
+            "queue_start_mjd": float,
+            "queue_fill_mjd_ns": np.int64,
+        },
```

Comment:
Is this going to fail the param checking if the columns don’t match exactly? I am undecided if that is good or bad, given that we may change the scheduler outputs and this would then tie the scheduler + schedview versions … OTOH, maybe that is just something to consider about the refactoring and the columns here could be tied to the columns created by the scheduler when it’s run?

Reply:
It will fail if the provided DataFrame doesn't have these columns, but it will pass if it has extras. I think this is what we want, because these are the columns the dashboard needs to function, and the dashboard will happily ignore extras.